From 8fba0b1726770ca77448268e96be1939027eb8af Mon Sep 17 00:00:00 2001
Message-Id: <8fba0b1726770ca77448268e96be1939027eb8af.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Sat, 27 Dec 2014 12:40:31 +0530
Subject: [PATCH 10/95] SQUASHME! [VPG]: drm/i915: Enable DPST after drm
 vblank on.

Since DPST uses vblank, we need to enable DPST post drm_vblank on this
was a ignored durong DPST enable on GMIN.

The initial Fixed DPST issues on GMIN patch was introduced in:
commit d96d2872ccf338a42086d5e068d2069c0409b914
Author: arsharma <ankitprasad.r.sharma@intel.com>
Date:   Fri Jul 4 15:09:11 2014 +0530

    MUST_REBASE [VPG]: drm/i915: Fixed DPST issues on GMIN

Issue: GMIN-3620
Change-Id: Iff0fb65d844674edbe287bcfda658969e66b3c7d
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index d847310..2d2d85e 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -5450,15 +5450,15 @@ static void valleyview_crtc_enable(struct drm_crtc *crtc)
 			encoder->enable(encoder);
 	}
 
-	/* Update DPST context after mode change */
-	if (I915_HAS_DPST(dev))
-		i915_dpst_display_on(dev);
-
 	intel_crtc_enable_planes(crtc);
 	intel_update_drrs(dev);
 
 	drm_crtc_vblank_on(crtc);
 
+	/* Update DPST context after mode change */
+	if (I915_HAS_DPST(dev))
+		i915_dpst_display_on(dev);
+
 	/* Underruns don't raise interrupts, so check manually. */
 	i9xx_check_fifo_underruns(dev);
 
-- 
1.7.9.5

